Quarkus 3.0.0.CR2 - quarkus-kafka-client triggers NativeImageBuildStep in JVM mode

### Describe the bug

Quarkus 3.0.0.CR2 - `quarkus-kafka-client` extension triggers NativeImageBuildStep in JVM mode

Regression comparing to 3.0.0.CR1, visible through this warning on LINUX machines

```
 [WARNING] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Cannot find the `native-image` in the 
  GRAALVM_HOME, JAVA_HOME and System PATH. Install it using `gu install native-image` 
  Attempting to fall back to container build.
```

Other extensions do not trigger this warnings. 

### Expected behavior

No warning from NativeImageBuildStep

### Actual behavior
```
 [WARNING] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Cannot find the `native-image` in the 
  GRAALVM_HOME, JAVA_HOME and System PATH. Install it using `gu install native-image` 
  Attempting to fall back to container build.
```

### How to Reproduce?

generate sample app using `mvn io.quarkus.platform:quarkus-maven-plugin:3.0.0.CR2:create -DprojectGroupId=my-groupId -DprojectArtifactId=app-quarkus-kafka-client -DprojectVersion=1.0.0-SNAPSHOT -DpackageName=org.my.group -DquarkusRegistryClient=false -DplatformVersion=3.0.0.CR2 -Dextensions=quarkus-kafka-client`

run `mvn clean package -DskipTests -f app-quarkus-kafka-client`

Check the log for the warning

CR1 has no warning on Linux machines using `mvn clean package -DskipTests -f app-quarkus-kafka-client -Dquarkus.platform.version=3.0.0.CR1` command

### Output of `uname -a` or `ver`

Linux

### Output of `java -version`

Java 11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.0.CR2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

We have a build here of a small quarkus application, and when trying to use CR2 instead of CR1 I also get the same warning. But as it is build inside an image, which does not have docker/podman inside it leads to a build error:
```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.0.0.CR2:build (default) on project qcer-distribution-app: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#resolveNativeImageBuildRunner threw an exception: java.lang.IllegalStateException: No container runtime was found. Make sure you have either Docker or Podman installed in your environment.
```
Currently trying to determine why he is taking the new route via the NativeImageBuildStep.

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/NativeImageRunnerBuildItem.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/NativeImageRunnerBuildItem.java
@@ -5,6 +5,11 @@ import io.quarkus.deployment.pkg.steps.NativeImageBuildRunner;
 
 /**
  * The resolved factory for the native image runner.
+ * <p>
+ * Warning: This build item should not be consumed without the use of
+ * {@link io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild} in the {@code onlyIf} method of
+ * {@link io.quarkus.builder.BuildStep} because that leads to Quarkus having to resolve the container image runtime
+ * unnecessarily.
  */
 public final class NativeImageRunnerBuildItem extends SimpleBuildItem {
     private final NativeImageBuildRunner buildRunner;

--- a/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java
+++ b/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java
@@ -6,6 +6,7 @@ import java.util.Collection;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
 import java.util.logging.Level;
 import java.util.stream.Collectors;
@@ -86,10 +87,13 @@ import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
 import io.quarkus.deployment.logging.LogCleanupFilterBuildItem;
 import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
 import io.quarkus.deployment.pkg.builditem.NativeImageRunnerBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeOrNativeSourcesBuild;
 import io.quarkus.dev.spi.DevModeType;
 import io.quarkus.devconsole.spi.DevConsoleRouteBuildItem;
 import io.quarkus.devconsole.spi.DevConsoleWebjarBuildItem;
-import io.quarkus.kafka.client.runtime.*;
+import io.quarkus.kafka.client.runtime.KafkaAdminClient;
+import io.quarkus.kafka.client.runtime.KafkaBindingConverter;
+import io.quarkus.kafka.client.runtime.KafkaRecorder;
 import io.quarkus.kafka.client.runtime.KafkaRuntimeConfigProducer;
 import io.quarkus.kafka.client.runtime.ui.KafkaTopicClient;
 import io.quarkus.kafka.client.runtime.ui.KafkaUiRecorder;
@@ -222,7 +226,6 @@ public class KafkaProcessor {
             BuildProducer<ServiceProviderBuildItem> serviceProviders,
             BuildProducer<NativeImageProxyDefinitionBuildItem> proxies,
             Capabilities capabilities,
-            NativeImageRunnerBuildItem nativeImageRunner,
             BuildProducer<UnremovableBeanBuildItem> beans,
             BuildProducer<NativeImageResourceBuildItem> nativeLibs,
             BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport) {
@@ -286,13 +289,11 @@ public class KafkaProcessor {
         handleAvro(reflectiveClass, proxies, serviceProviders, sslNativeSupport, capabilities);
         handleOpenTracing(reflectiveClass, capabilities);
         handleStrimziOAuth(curateOutcomeBuildItem, reflectiveClass);
-        if (config.snappyEnabled) {
-            handleSnappy(nativeImageRunner, reflectiveClass, nativeLibs);
-        }
 
     }
 
-    private void handleSnappy(NativeImageRunnerBuildItem nativeImageRunner,
+    @BuildStep(onlyIf = { IsSnappy.class, NativeOrNativeSourcesBuild.class })
+    public void handleSnappyInNative(NativeImageRunnerBuildItem nativeImageRunner,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             BuildProducer<NativeImageResourceBuildItem> nativeLibs) {
         reflectiveClass.produce(ReflectiveClassBuildItem.builder("org.xerial.snappy.SnappyInputStream",
@@ -572,6 +573,20 @@ public class KafkaProcessor {
                 .root(DEVCONSOLE_WEBJAR_STATIC_RESOURCES_PATH)
                 .routeRoot(KAFKA_RESOURCES_ROOT_PATH)
                 .build();
+    }
+
+    public static final class IsSnappy implements BooleanSupplier {
+
+        private final KafkaBuildTimeConfig config;
+
+        public IsSnappy(KafkaBuildTimeConfig config) {
+            this.config = config;
+        }
+
+        @Override
+        public boolean getAsBoolean() {
+            return config.snappyEnabled;
+        }
     }
 
 }

